src/main/java/simulation/Simulation.java

```diff
@@ -1,85 +1,64 @@
 import boards.WhompsDominoRuins;
-import boards.spaces.BaseSpace;
-import mg.MinigameManager;
-import org.apache.commons.collections4.CollectionUtils;
 import org.apache.commons.lang3.time.StopWatch;
-import partydice.BobombAlly;
 import partydice.Dice;
-import results.CoinResult;
-import results.DieResult;
-import results.MoveResult;
-import stattracker.AllyStatTracker;
-import stattracker.GameStatTracker;
-import stattracker.SimulationStatTracker;
-import utils.RandomUtils;
 
 import java.io.BufferedWriter;
 import java.io.FileWriter;
 import java.io.IOException;
 import java.text.DecimalFormat;
 import java.time.LocalDateTime;
-import java.util.ArrayList;
-import java.util.Collections;
-import java.util.List;
+import java.util.Arrays;
+import java.util.concurrent.CompletableFuture;
+import java.util.function.Supplier;
 
-public class Simulation implements Runnable{
+public class Simulation implements Runnable {
     protected final DecimalFormat DECIMAL_FORMAT = new DecimalFormat("####.#######");
 
-    protected static final int TURN_COUNT = 20;
-    protected int SIM_COUNT = 6000000;
+    protected int SIM_COUNT;
 
-    protected BaseBoard gameBoard;
-
-    protected SimulationStatTracker simulationStatTracker;
+    protected Supplier<BaseBoard> gameBoard;
 
     protected String fileOutputName;
 
-    protected MinigameManager minigameManager;
-
     public Simulation() {
-        this.gameBoard = new WhompsDominoRuins();
+        this.gameBoard = WhompsDominoRuins::new;
     }
 
-    public Simulation(BaseBoard gameBoard, int simCount) {
+    public Simulation(Supplier<BaseBoard> gameBoard, String fileOutputName, int simCount) {
         SIM_COUNT = simCount;
         this.gameBoard = gameBoard;
-        fileOutputName = "output/" + gameBoard.getFileOutputName();
-        minigameManager = new MinigameManager();
+        this.fileOutputName = "output/" + fileOutputName;
     }
 
     @Override
     public void run() {
-        simulate();
-    }
-
-    public void simulate() {
         StopWatch stopWatch = StopWatch.createStarted();
-        System.out.println(gameBoard.getFileOutputName() + " time start!\t\t" + LocalDateTime.now());
+        System.out.println(fileOutputName + " time start!\t\t" + LocalDateTime.now());
         printTableHeaders();
 
-        for (Dice characterDie : Dice.values()) {
-            simulationStatTracker = new SimulationStatTracker(characterDie);
+        CompletableFuture<?>[] characterSims = Arrays.stream(Dice.values())
+                .map(characterDie -> new CharacterSimulation(gameBoard.get(), characterDie, SIM_COUNT))
+                .map(CompletableFuture::runAsync).toList()
+                .toArray(CompletableFuture<?>[]::new);
 
-            for (int i = 0; i < SIM_COUNT; ++i) {
-                simulateGame();
-                simulationStatTracker.endGame();
-                gameBoard.resetBoard();
+        try {
+            while(!CompletableFuture.allOf(characterSims).isDone()) {

```

Comment:
Consider using `characterSims.forEach(CompletableFuture::join)` to wait for each thread to signal completion, rather than a sleep loop.

I'm not as familiar with CompletableFuture, so there might be a reason I'm missing to do it this way - but if the goal is just to wait for them all to complete, that is supported natively by the framework.
Here's a good StackOverflow discussion on waiting for CompletableFutures that I found helpful:
https://stackoverflow.com/questions/30705981/what-is-the-recommended-way-to-wait-till-the-completable-future-threads-finish